### ISSUE TITLE
Question about epoch setting for CLEVRTex reproduction

### ISSUE THREAD

**Issue by 0930mcx** [Wuziyi616/SlotDiffusion#8]
Thank you very much for your excellent work. I hope you don't mind me reaching out with a small question regarding the replication of your results. When training SADiffusion on the CLEVRTex dataset, I noticed that you mentioned using 100 epochs in the paper. However, in the default configuration file sa_ldm_clevrtex_params-res128.py, it is set to 400 epochs. Could you kindly advise on what settings I should use to accurately reproduce the results reported in your paper? I'd really appreciate your guidance. Thank you!

**Comment by Wuziyi616** [comment]
Hi, it's been a long time, so honestly I don't remember. I think the `100 epochs` in the paper should be the correct number though. When I did experiments, I often trained the model for long enough, and picked the checkpoint that has the lowest validation loss to compute metrics. So the number of epochs in the training config might be too long.

**Comment by 0930mcx** [comment]
Thank you for your previous response. I'm now encountering some discrepancies between my reproduced results and those reported in your paper. I noticed that your default configuration uses single-GPU training, whereas I have been using two 4090 GPUs with DDP. Could this have an impact on the results?

I also observed that the `nerv` repository had a bug fix related to DDP, which was implemented after the version 0.2.0 recommended in your `install.md`. If I use the recommended version of `nerv`, might it still trigger the DDP bug and lead to degraded training performance? If so, which version of `nerv` should I use to correctly reproduce your results?

I would really appreciate your guidance on this. Thank you!

**Comment by Wuziyi616** [comment]
Yes, it does make a difference, please use the new version of `nerv`. Before the DDP bug fix, when you train with 2 GPUs, the code actually ignores the gradient update from the second GPU, and only uses the first GPU. That leads to a halved effective batch size so should makes the performance worse. Please try the updated package and let me know if it helps.

**Comment by 0930mcx** [comment]
I have now updated `nerv` to the latest version, but I'm still encountering issues during reproduction. My configuration file is as follows:

[sa_ldm_dino_voc_params-res224.py](https://github.com/user-attachments/files/26154836/sa_ldm_dino_voc_params-res224.py)

**Comment by Wuziyi616** [comment]
Did you change anything other than `gpus`? What are the results you get vs the results in my paper?

**Comment by 0930mcx** [comment]
I also modified the `VOCTransforms` class in `img_based/datasets/voc_transform.py` by adding the following line:
```python
self.normalize = Normalize(norm_mean, norm_std)
```
The reason is that `method.py` contains the following calls:
```python
norm = dst.__getattribute__(f'{dst_name}_transforms').normalize
img = norm.denormalize_image(img)
```
In the original `voc_transform.py`, the `normalize` object was not defined, so I added it to avoid errors.

With these modifications, the results I obtained are mBO = 42.5 and FARI = 15.5.

**Comment by Wuziyi616** [comment]
This is my training curve for your reference. How many steps did you train for? Maybe 100 epochs is too few and you should do 400. From the wandb plot, I trained for ~45k steps.

Also note that as an unsupervised learning method, there are variations across different runs due to e.g. random seeds. Usually I train the model with 2 seeds and pick the better one.

Training loss:
<img width="2583" height="447" alt="Image" src="https://github.com/user-attachments/assets/16473774-300b-4bfd-8622-b214635c1990" />

Eval:
<img width="1701" height="914" alt="Image" src="https://github.com/user-attachments/assets/a7d4182c-aaeb-4b79-b72c-82e5ad9b162b" />

**Comment by Wuziyi616** [comment]
Feel free to reopen if you have more questions in reproducing the result!
